File: tests/fixtures/eval_ts/src/lib/queue.ts
export interface Job {
  id: string;
  priority: number;
}

export class Queue {
  private jobs: Job[] = [];

  push(job: Job): void {
    this.jobs.push(job);
  }

  pop(): Job | undefined {
    return this.jobs.shift();
  }

  size(): number {
    return this.jobs.length;
  }
}

export function drainQueue(queue: Queue): Job[] {
  const out: Job[] = [];
  let next = queue.pop();
  while (next) {
    out.push(next);
    next = queue.pop();
  }
  return out;
}

File: tests/fixtures/eval_py/pkg/queue.py
from pkg.types import Job


class Queue:
    def __init__(self) -> None:
        self._jobs: list[Job] = []

    def push(self, job: Job) -> None:
        self._jobs.append(job)

    def pop(self) -> Job | None:
        if not self._jobs:
            return None
        return self._jobs.pop(0)


def drain_queue(queue: Queue) -> list[Job]:
    out: list[Job] = []
    while True:
        job = queue.pop()
        if job is None:
            break
        out.append(job)
    return out
